// @prettier/plugin-pug/dist/options/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PUG_BRACKET_SAME_LINE_OPTION = exports.PUG_ARROW_PARENS_OPTION = exports.PUG_SEMI_OPTION = exports.PUG_BRACKET_SPACING_OPTION = exports.PUG_USE_TABS_OPTION = exports.PUG_TAB_WIDTH_OPTION = exports.PUG_SINGLE_QUOTE_OPTION = exports.PUG_PRINT_WIDTH_OPTION = void 0;
const constants_1 = require("./constants");
exports.PUG_PRINT_WIDTH_OPTION = {
    since: '1.6.0',
    category: constants_1.CATEGORY_PUG,
    type: 'int',
    default: -1,
    description: 'The line length where Prettier will try wrap.',
    range: { start: -1, end: Infinity, step: 1 },
};
exports.PUG_SINGLE_QUOTE_OPTION = {
    since: '1.6.0',
    category: constants_1.CATEGORY_PUG,
    type: 'choice',
    default: null,
    description: '',
    choices: [
        {
            value: null,
            description: 'Use `singleQuote` value.',
        },
        {
            value: true,
            description: 'Use single quotes instead of double quotes.',
        },
        {
            value: 'true',
            description: 'Use single quotes instead of double quotes.',
        },
        {
            value: false,
            description: 'Use double quotes instead of double quotes.',
        },
    ],
};
exports.PUG_TAB_WIDTH_OPTION = {
    since: '1.6.0',
    category: constants_1.CATEGORY_PUG,
    type: 'int',
    default: -1,
    description: 'Number of spaces per indentation level.',
    range: { start: -1, end: Infinity, step: 1 },
};
exports.PUG_USE_TABS_OPTION = {
    since: '1.6.0',
    category: constants_1.CATEGORY_PUG,
    type: 'choice',
    default: null,
    description: '',
    choices: [
        {
            value: null,
            description: 'Use `useTabs` value.',
        },
        {
            value: true,
            description: 'Indent with tabs instead of spaces.',
        },
        {
            value: 'true',
            description: 'Indent with tabs instead of spaces.',
        },
        {
            value: false,
            description: 'Indent with spaces instead of tabs.',
        },
    ],
};
exports.PUG_BRACKET_SPACING_OPTION = {
    since: '1.6.0',
    category: constants_1.CATEGORY_PUG,
    type: 'choice',
    default: null,
    description: '',
    choices: [
        {
            value: null,
            description: 'Use `bracketSpacing` value.',
        },
        {
            value: true,
            description: 'Print spaces between brackets.',
        },
        {
            value: 'true',
            description: 'Print spaces between brackets.',
        },
        {
            value: false,
            description: 'Do not print spaces between brackets.',
        },
    ],
};
exports.PUG_SEMI_OPTION = {
    since: '1.6.0',
    category: constants_1.CATEGORY_PUG,
    type: 'choice',
    default: null,
    description: '',
    choices: [
        {
            value: null,
            description: 'Use `bracketSpacing` value.',
        },
        {
            value: true,
            description: 'Print semicolons.',
        },
        {
            value: 'true',
            description: 'Print semicolons.',
        },
        {
            value: false,
            description: 'Do not print semicolons, except at the beginning of lines which may need them.',
        },
    ],
};
exports.PUG_ARROW_PARENS_OPTION = {
    since: '1.7.0',
    category: constants_1.CATEGORY_PUG,
    type: 'choice',
    default: null,
    description: 'Include parentheses around a sole arrow function parameter.',
    choices: [
        {
            value: null,
            description: 'Use `arrowParens` value.',
        },
        {
            value: 'always',
            description: 'Always add parens. Example: `(x) => x`',
        },
        {
            value: 'avoid',
            description: 'Omit parens when possible. Example: `x => x`',
        },
    ],
};
exports.PUG_BRACKET_SAME_LINE_OPTION = {
    since: '1.17.0',
    category: constants_1.CATEGORY_PUG,
    type: 'choice',
    default: null,
    description: 'Determines position of closing bracket which wraps attributes.',
    choices: [
        {
            value: null,
            description: 'Use `bracketSameLine` value.',
        },
        {
            value: true,
            description: `
			Closing bracket remains with last attribute's line.
			Example:
			input(
				type='text',
				value='my_value',
				name='my_name',
				alt='my_alt',
				autocomplete='on')
			`,
        },
        {
            value: 'true',
            description: `
			Closing bracket remains with last attribute's line.
			Example:
			input(
				type='text',
				value='my_value',
				name='my_name',
				alt='my_alt',
				autocomplete='on')
			`,
        },
        {
            value: false,
            description: `
			Closing bracket ends with a new line.
			Example:
			input(
				type='text',
				value='my_value',
				name='my_name',
				alt='my_alt',
				autocomplete='on'
			)
			`,
        },
    ],
};
