// @prettier/plugin-pug/dist/options/converge.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.convergeOptions = void 0;
function convergeOptions(options) {
    return {
        printWidth: options.printWidth,
        pugPrintWidth: options.pugPrintWidth !== -1 ? options.pugPrintWidth : options.printWidth,
        singleQuote: options.singleQuote,
        pugSingleQuote: options.pugSingleQuote ?? options.singleQuote,
        tabWidth: options.tabWidth,
        pugTabWidth: options.pugTabWidth !== -1 ? options.pugTabWidth : options.tabWidth,
        useTabs: options.useTabs,
        pugUseTabs: options.pugUseTabs ?? options.useTabs,
        bracketSpacing: options.bracketSpacing,
        pugBracketSpacing: options.pugBracketSpacing ?? options.bracketSpacing,
        arrowParens: options.arrowParens,
        pugArrowParens: options.pugArrowParens ?? options.arrowParens,
        semi: options.semi,
        pugSemi: options.pugSemi ?? options.semi,
        bracketSameLine: options.bracketSameLine,
        pugBracketSameLine: options.pugBracketSameLine ?? options.bracketSameLine,
        pugAttributeSeparator: options.pugAttributeSeparator,
        pugCommentPreserveSpaces: options.pugCommentPreserveSpaces,
        pugSortAttributes: options.pugSortAttributes,
        pugSortAttributesBeginning: options.pugSortAttributesBeginning,
        pugSortAttributesEnd: options.pugSortAttributesEnd,
        pugWrapAttributesThreshold: options.pugWrapAttributesThreshold,
        pugWrapAttributesPattern: options.pugWrapAttributesPattern,
        pugClassNotation: options.pugClassNotation,
        pugIdNotation: options.pugIdNotation,
        pugClassLocation: options.pugClassLocation,
        pugEmptyAttributes: options.pugEmptyAttributes,
        pugEmptyAttributesForceQuotes: options.pugEmptyAttributesForceQuotes,
        pugSingleFileComponentIndentation: options.pugSingleFileComponentIndentation &&
            options.__embeddedInHtml === true,
        pugFramework: options.pugFramework,
        pugExplicitDiv: options.pugExplicitDiv,
        pugPreserveAttributeBrackets: options.pugPreserveAttributeBrackets,
    };
}
exports.convergeOptions = convergeOptions;
