// @prettier/plugin-pug/dist/options/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CATEGORY_PUG = exports.options = void 0;
const attribute_sorting_1 = require("./attribute-sorting");
const common_1 = require("./common");
const empty_attributes_1 = require("./empty-attributes");
const pug_attribute_separator_1 = require("./pug-attribute-separator");
const pug_class_location_1 = require("./pug-class-location");
const pug_class_notation_1 = require("./pug-class-notation");
const pug_comment_preserve_spaces_1 = require("./pug-comment-preserve-spaces");
const pug_explicit_div_1 = require("./pug-explicit-div");
const pug_framework_1 = require("./pug-framework");
const pug_id_notation_1 = require("./pug-id-notation");
const pug_preserve_attribute_brackets_1 = require("./pug-preserve-attribute-brackets");
const pug_single_file_component_indentation_1 = require("./pug-single-file-component-indentation");
const pug_wrap_attributes_1 = require("./pug-wrap-attributes");
exports.options = {
    pugPrintWidth: common_1.PUG_PRINT_WIDTH_OPTION,
    pugSingleQuote: common_1.PUG_SINGLE_QUOTE_OPTION,
    pugTabWidth: common_1.PUG_TAB_WIDTH_OPTION,
    pugUseTabs: common_1.PUG_USE_TABS_OPTION,
    pugBracketSpacing: common_1.PUG_BRACKET_SPACING_OPTION,
    pugArrowParens: common_1.PUG_ARROW_PARENS_OPTION,
    pugSemi: common_1.PUG_SEMI_OPTION,
    pugBracketSameLine: common_1.PUG_BRACKET_SAME_LINE_OPTION,
    pugAttributeSeparator: pug_attribute_separator_1.PUG_ATTRIBUTE_SEPARATOR_OPTION,
    pugCommentPreserveSpaces: pug_comment_preserve_spaces_1.PUG_COMMENT_PRESERVE_SPACES_OPTION,
    pugSortAttributes: attribute_sorting_1.PUG_SORT_ATTRIBUTES_OPTION,
    pugSortAttributesBeginning: attribute_sorting_1.PUG_SORT_ATTRIBUTES_BEGINNING_OPTION,
    pugSortAttributesEnd: attribute_sorting_1.PUG_SORT_ATTRIBUTES_END_OPTION,
    pugWrapAttributesThreshold: pug_wrap_attributes_1.PUG_WRAP_ATTRIBUTES_THRESHOLD,
    pugWrapAttributesPattern: pug_wrap_attributes_1.PUG_WRAP_ATTRIBUTES_PATTERN,
    pugEmptyAttributes: empty_attributes_1.PUG_EMPTY_ATTRIBUTES_OPTION,
    pugClassNotation: pug_class_notation_1.PUG_CLASS_NOTATION,
    pugIdNotation: pug_id_notation_1.PUG_ID_NOTATION,
    pugClassLocation: pug_class_location_1.PUG_CLASS_LOCATION,
    pugExplicitDiv: pug_explicit_div_1.PUG_EXPLICIT_DIV,
    pugEmptyAttributesForceQuotes: empty_attributes_1.PUG_EMPTY_ATTRIBUTES_FORCE_QUOTES_OPTION,
    pugSingleFileComponentIndentation: pug_single_file_component_indentation_1.PUG_SINGLE_FILE_COMPONENT_INDENTATION,
    pugFramework: pug_framework_1.PUG_FRAMEWORK,
    pugPreserveAttributeBrackets: pug_preserve_attribute_brackets_1.PUG_PRESERVE_ATTRIBUTE_BRACKETS,
};
var constants_1 = require("./constants");
Object.defineProperty(exports, "CATEGORY_PUG", { enumerable: true, get: function () { return constants_1.CATEGORY_PUG; } });
