// @prettier/plugin-pug/dist/utils/vue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isVueVDirective = exports.isVueVOnExpression = exports.isVueVBindExpression = exports.isVueVForWithOf = exports.isVueExpression = exports.isVueEventBinding = void 0;
function isVueEventBinding(name) {
    return /^(v-on:|@).*/.test(name);
}
exports.isVueEventBinding = isVueEventBinding;
function isVueExpression(name) {
    return /^((v-(bind|slot))?:|v-(model|slot|if|for|else-if|text|html|t)|#).*/.test(name);
}
exports.isVueExpression = isVueExpression;
function isVueVForWithOf(name, val) {
    return 'v-for' === name && val.includes('of');
}
exports.isVueVForWithOf = isVueVForWithOf;
function isVueVBindExpression(name) {
    return 'v-bind' === name;
}
exports.isVueVBindExpression = isVueVBindExpression;
function isVueVOnExpression(name) {
    return 'v-on' === name;
}
exports.isVueVOnExpression = isVueVOnExpression;
function isVueVDirective(name) {
    return name.startsWith('v-');
}
exports.isVueVDirective = isVueVDirective;
